<Type Name="UnhandledExceptionAction" FullName="System.Activities.UnhandledExceptionAction">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="070ae82d795d2b3a920f23bb6d8e67539cdf87c2" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="zh-TW" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
  </Metadata>
  <TypeSignature Language="C#" Value="public enum UnhandledExceptionAction" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed UnhandledExceptionAction extends System.Enum" />
  <TypeSignature Language="DocId" Value="T:System.Activities.UnhandledExceptionAction" />
  <TypeSignature Language="VB.NET" Value="Public Enum UnhandledExceptionAction" />
  <TypeSignature Language="C++ CLI" Value="public enum class UnhandledExceptionAction" />
  <AssemblyInfo>
    <AssemblyName>System.Activities</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Enum</BaseTypeName>
  </Base>
  <Docs>
    <summary>
      <span data-ttu-id="89cca-101">指定在例外狀況溢出工作流程的根時出現的動作。</span>
      <span class="sxs-lookup">
        <span data-stu-id="89cca-101">Specifies the action that occurs when an exception escapes the root of a workflow.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="89cca-102">當例外狀況逸出工作流程的根時，會叫用 <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> 函式。</span><span class="sxs-lookup"><span data-stu-id="89cca-102">The <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> function is invoked if an exception escapes the root of the workflow.</span></span> <span data-ttu-id="89cca-103"><xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> 會提供此例外狀況的存取權，以及產生例外狀況之 <xref:System.Activities.Activity> 的指標。</span><span class="sxs-lookup"><span data-stu-id="89cca-103">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> provides access to the exception as well as a pointer to the <xref:System.Activities.Activity> that generated the exception.</span></span> <span data-ttu-id="89cca-104">終止是預設的動作，如果沒有<xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>指定處理常式。</span><span class="sxs-lookup"><span data-stu-id="89cca-104">Terminate is the default action if no <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is specified.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="89cca-105">下列範例會叫用擲回例外狀況的工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-105">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="89cca-106">此例外狀況未由工作流程處理，而且叫用了 <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> 處理常式。</span><span class="sxs-lookup"><span data-stu-id="89cca-106">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="89cca-107">系統會檢查 <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> 以提供例外狀況的相關資訊，並且終止工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-107">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#6](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#6)]  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <Member MemberName="Abort">
      <MemberSignature Language="C#" Value="Abort" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Abort = int32(0)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Abort" />
      <MemberSignature Language="VB.NET" Value="Abort" />
      <MemberSignature Language="C++ CLI" Value="Abort" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>0</MemberValue>
      <Docs>
        <summary>
          <span data-ttu-id="89cca-108">指定 <see cref="T:System.Activities.WorkflowApplication" /> 應該中止工作流程。</span>
          <span class="sxs-lookup">
            <span data-stu-id="89cca-108">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should abort the workflow.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="89cca-109">這會導致系統在中止處理序完成時，呼叫 <xref:System.Activities.WorkflowApplication.Aborted%2A>。</span><span class="sxs-lookup"><span data-stu-id="89cca-109">This results in <xref:System.Activities.WorkflowApplication.Aborted%2A> being called when the abort process is complete.</span></span> <span data-ttu-id="89cca-110">使用未處理的例外狀況做為中止原因。</span><span class="sxs-lookup"><span data-stu-id="89cca-110">The unhandled exception is used as the abort reason.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="89cca-111">下列範例會叫用擲回例外狀況的工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-111">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="89cca-112">此例外狀況未由工作流程處理，而且叫用了 <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> 處理常式。</span><span class="sxs-lookup"><span data-stu-id="89cca-112">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="89cca-113">系統會檢查 <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> 以提供例外狀況的相關資訊，並且中止工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-113">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is aborted.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#7](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#7)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Cancel">
      <MemberSignature Language="C#" Value="Cancel" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Cancel = int32(1)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Cancel" />
      <MemberSignature Language="VB.NET" Value="Cancel" />
      <MemberSignature Language="C++ CLI" Value="Cancel" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>1</MemberValue>
      <Docs>
        <summary>
          <span data-ttu-id="89cca-114">指定 <see cref="T:System.Activities.WorkflowApplication" /> 應該排程根活動的取消和繼續執行。</span>
          <span class="sxs-lookup">
            <span data-stu-id="89cca-114">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should schedule the cancellation of the root activity and resume execution.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="89cca-115">這會導致系統在取消處理序完成時，呼叫 <xref:System.Activities.WorkflowApplication.Completed%2A>。</span><span class="sxs-lookup"><span data-stu-id="89cca-115">This results in <xref:System.Activities.WorkflowApplication.Completed%2A> being called when the cancellation process is complete.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="89cca-116">下列範例會叫用擲回例外狀況的工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-116">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="89cca-117">此例外狀況未由工作流程處理，而且叫用了 <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> 處理常式。</span><span class="sxs-lookup"><span data-stu-id="89cca-117">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="89cca-118">系統會檢查 <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> 以提供例外狀況的相關資訊，並且取消工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-118">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is cancelled.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#8](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#8)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Terminate">
      <MemberSignature Language="C#" Value="Terminate" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Terminate = int32(2)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Terminate" />
      <MemberSignature Language="VB.NET" Value="Terminate" />
      <MemberSignature Language="C++ CLI" Value="Terminate" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>2</MemberValue>
      <Docs>
        <summary>
          <span data-ttu-id="89cca-119">指定 <see cref="T:System.Activities.WorkflowApplication" /> 應該排程根活動的終止和繼續執行。</span>
          <span class="sxs-lookup">
            <span data-stu-id="89cca-119">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should schedule termination of the root activity and resume execution.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="89cca-120">當終止處理序完成時，呼叫 <xref:System.Activities.WorkflowApplication.Completed%2A> 中的此結果。</span><span class="sxs-lookup"><span data-stu-id="89cca-120">This results in <xref:System.Activities.WorkflowApplication.Completed%2A> being called when the termination process is complete.</span></span> <span data-ttu-id="89cca-121">使用未處理的例外狀況做為終止原因。</span><span class="sxs-lookup"><span data-stu-id="89cca-121">The unhandled exception is used as the termination reason.</span></span> <span data-ttu-id="89cca-122">終止是預設的動作，如果沒有<xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>指定處理常式。</span><span class="sxs-lookup"><span data-stu-id="89cca-122">Terminate is the default action if no <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is specified.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="89cca-123">下列範例會叫用擲回例外狀況的工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-123">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="89cca-124">此例外狀況未由工作流程處理，而且叫用了 <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> 處理常式。</span><span class="sxs-lookup"><span data-stu-id="89cca-124">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="89cca-125">系統會檢查 <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> 以提供例外狀況的相關資訊，並且終止工作流程。</span><span class="sxs-lookup"><span data-stu-id="89cca-125">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#6](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#6)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>